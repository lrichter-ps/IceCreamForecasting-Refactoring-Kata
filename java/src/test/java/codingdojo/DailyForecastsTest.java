package codingdojo;

import org.approvaltests.Approvals;
import org.approvaltests.core.Options;
import org.approvaltests.scrubbers.DateScrubber;
import org.approvaltests.scrubbers.RegExScrubber;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

public class DailyForecastsTest {

    final ByteArrayOutputStream capturedOutput = new ByteArrayOutputStream();
    PrintStream originalOutput;

    @BeforeEach
    void captureStandardOutput() {
        originalOutput = System.out;
        System.setOut(new PrintStream(capturedOutput));
    }

    @AfterEach
    void resetStandardOutput() {
        System.setOut(originalOutput);
    }

    @Test
    void salesForecast() {
        var forecasts = new DailyForecasts();
        forecasts.printSalesForecasts();
        Approvals.verify(capturedOutput);
    }
}
