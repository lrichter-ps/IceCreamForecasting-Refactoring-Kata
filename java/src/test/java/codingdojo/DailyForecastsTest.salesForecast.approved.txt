Forecast at: date
Steve score: 9
Julie score: 11
Francis score: 9
